Britannia/common/buildings/BTWK_special_buildings.txt

```diff
@@ -0,0 +1,175 @@
+﻿##########
+# Hadrian's Wall
+##########
+
+hadrians_wall_01 = {
+	# Hadrian's Wall - Ruined
+	# Change province modifiers?
+	# Cannot be restored from here - totally ruined.
+	flag = fully_upgraded
+	type_icon = "icon_structure_hadrians_wall.dds"
+	
+
+	province_modifier = {
+		build_gold_cost = -0.1
+	}
+
+	ai_value = {
+		base = 100
+		modifier = { # Fill all building slots before going for special buildings
+			factor = 0
+			free_building_slots > 0
+		}
+	}
+	type = special
+}
+
+hadrians_wall_02 = {
+	# Can't be constructed
+	# Hadrian's Wall 
+
+	type_icon = "icon_structure_hadrians_wall.dds"
+
+	province_modifier = {
+		fort_level = 1
+		defender_holding_advantage = 5
+		build_gold_cost = -0.025
+	}
+	next_building = hadrians_wall_03
+	type = special
+}
+
+hadrians_wall_03 = {
+	# Hadrian's Wall - Restored
+	flag = fully_upgraded
+	construction_time = slow_construction_time
+	cost_gold = 500
+	type_icon = "icon_structure_hadrians_wall.dds"
+	
+	can_construct = {
+		culture = { 
+			has_innovation = innovation_crop_rotation
+		}
+		scope:holder.top_liege = {
+			is_imperial_like_culture_trigger = yes 
+				completely_controls_region = custom_hadrians_wall_duchies
+				OR = {
```

Comment:
This is partly duplicate with is_imperial_like_culture_trigger = yes just above - not sure which one is intended, probably this one?